libbpf-rs/src/perf_buffer.rs

```diff
@@ -7,44 +7,39 @@ use crate::*;
 pub struct PerfBuffer {}
 
 impl PerfBuffer {
-    pub fn new(_map: Map, _pages: usize, _opts: PerfBufferOpts) -> Self {
-        unimplemented!();
-    }
-
-    pub fn poll(&mut self, _timeout: Duration) {
-        unimplemented!();
-    }
-}
-
-/// Options and callbacks to configure [`PerfBuffer`].
-///
-/// Some callbacks provide raw byte slices. You may find libraries such as
-/// [`plain`](https://crates.io/crates/plain) helpful.
-#[derive(Default)]
-pub struct PerfBufferOpts {}
-
-impl PerfBufferOpts {
-    pub fn new() -> Self {
+    pub fn new(_map: Map) -> Self {
```

Comment:
same as above, I think using map_fd here is more flexible and easier in some cases for user code. Plus, I'd imagine you'd pass LoadedMap here, instead of not-yet-constructred Map, which doesn't yet even have fd

Reply:
Let's go with `Map` (post-rename from `LoadedMap` -> `Map`) for now. We can add a `from_fd()` method when we have more FD based APIs.

Reply:
yeah, was thinking about Map::from_fd() constructor, you'll need to do GET_OBJ_INFO to get back map name, key/value sizes, etc, but it should work as well.